From: Carlos Garnacho <carlosg@gnome.org>
Date: Sat, 23 Apr 2022 19:53:46 +0200
Subject: padOsd: Set menu manager on pad chooser popup

Likewise to action menus, this does now need a menu manager
that lets the popup stack a grab on top of the pad OSD's.

Part-of: <https://gitlab.gnome.org/GNOME/gnome-shell/-/merge_requests/2279>
Origin: upstream, 42.1, commit:9ee7af233cabae3a1708ab655b235fd0db829cc3
---
 js/ui/padOsd.js | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/js/ui/padOsd.js b/js/ui/padOsd.js
index e23213c..b273b0c 100644
--- a/js/ui/padOsd.js
+++ b/js/ui/padOsd.js
@@ -67,6 +67,9 @@ var PadChooser = GObject.registerClass({
         this._padChooserMenu.actor.hide();
         Main.uiGroup.add_actor(this._padChooserMenu.actor);
 
+        this._menuManager = new PopupMenu.PopupMenuManager(this);
+        this._menuManager.addMenu(this._padChooserMenu);
+
         for (let i = 0; i < devices.length; i++) {
             let device = devices[i];
             if (device == this.currentDevice)
